fiftyone/utils/hf_hub.py

```diff
@@ -0,0 +1,1109 @@
+from concurrent.futures import ThreadPoolExecutor
+import json
+import logging
+import os
+from packaging.requirements import Requirement
+from PIL import Image
+import requests
+import shutil
+
+import huggingface_hub as hfh
+import yaml
+
+import fiftyone as fo
+import fiftyone.constants as foc
+from fiftyone.core.config import Config
+import fiftyone.core.dataset as fod
+import fiftyone.core.labels as fol
+import fiftyone.core.metadata as fom
+from fiftyone.core.sample import Sample
+import fiftyone.core.utils as fou
+import fiftyone.types as fot
+
+DATASETS_SERVER_URL = "https://datasets-server.huggingface.co"
+DEFAULT_MEDIA_TYPE = "image"
+DATASET_METADATA_FILENAMES = ("fiftyone.yml", "fiftyone.yaml")
+DATASETS_MAX_BATCH_SIZE = 100
+DEFAULT_IMAGE_FILEPATH_FEATURE = "image"
+FIFTYONE_BUILTIN_FIELDS = ("id", "filepath", "tags", "metadata")
+REPO_TYPE = "dataset"
+SUPPORTED_DTYPES = (
+    "int8",
+    "int16",
+    "int32",
+    "int64",
+    "float16",
+    "float32",
+    "float64",
+    "bool",
+    "string",
+)
+
+logger = logging.getLogger(__name__)
+
+
+def get_cache():
```

Comment:
@jacobmarks ah I missed this before; can you refactor to avoid using `globals`? You can likely do this by passing a list between your various functions and append new paths to it.

Using globals is not recommended in general as it can cause issues with concurrency. Also in this implementation, if a user presses ctrl+c during execution, I believe the global won't be cleared and then the next time they try to download a dataset they'll get the old files downloaded as an unintended surprise!

Reply:
Done!